launcher/ui/MainWindow.cpp

```diff
@@ -1495,141 +1495,163 @@ void MainWindow::on_actionCreateInstanceShortcut_triggered()
 {
     if (!m_selectedInstance)
         return;
-    auto desktopPath = FS::getDesktopDir();
-    if (desktopPath.isEmpty()) {
-        // TODO come up with an alternative solution (open "save file" dialog)
-        QMessageBox::critical(this, tr("Create instance shortcut"), tr("Couldn't find desktop?!"));
-        return;
+
+    std::vector<QString> paths;
+    QString mode = APPLICATION->settings()->get("ShortcutCreationMode").toString();
+    if (mode == "Applications") {
+        paths.push_back(FS::getApplicationsDir());
+    } else if (mode == "Both") {
+        paths.push_back(FS::getDesktopDir());
+        paths.push_back(FS::getApplicationsDir());
+    } else {
+        // Default to desktop
+        paths.push_back(FS::getDesktopDir());
     }
 
-    QString desktopFilePath;
-    QString appPath = QApplication::applicationFilePath();
-    QString iconPath;
-    QStringList args;
+    for (const QString& shortcutDirPath : paths) {
+        if (shortcutDirPath.isEmpty()) {
```

Comment:
I know that this was not there before, but can't we just ask the user where he or she wants to save the desktop file (similar to Flatpak approach)?

Reply:
Maybe give them options like Desktop, Applications and other (where the save file dialog will be displayed)?